cli/internal/daemon/status.go

```diff
@@ -0,0 +1,98 @@
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/daemon/connector"
+	"github.com/vercel/turborepo/cli/internal/daemonclient"
+	"github.com/vercel/turborepo/cli/internal/fs"
+)
+
+func addStatusCmd(root *cobra.Command, config *config.Config, output cli.Ui) {
+	var outputJSON bool
+	cmd := &cobra.Command{
+		Use:           "status",
+		Short:         "Reports the status of the turbo daemon",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := &status{
+				logger:       config.Logger,
+				output:       output,
+				turboVersion: config.TurboVersion,
+				repoRoot:     config.Cwd,
+				outputJSON:   outputJSON,
+			}
+			return s.status()
+		},
+	}
+	cmd.Flags().BoolVar(&outputJSON, "json", false, "Pass --json to report status in JSON format")
+	root.AddCommand(cmd)
+}
+
+type status struct {
+	logger       hclog.Logger
+	output       cli.Ui
+	turboVersion string
+	repoRoot     fs.AbsolutePath
+	outputJSON   bool
+}
+
+func (s *status) status() error {
+	ctx := context.Background()
+	client, err := GetClient(ctx, s.repoRoot, s.logger, s.turboVersion, ClientOpts{
+		DontStart: true,
```

Comment:
Something somewhere is going to end up having the bag of options. Either here, at the public interface to the package, or down a level where the two variants would funnel into the same connection logic. 

I'd prefer to leave it here, since that reduces the surface area if we need to add another option (configurable idle time for an example of one we should consider).

The options-struct is a reasonably common Go pattern, and the next-most-common pattern is the option-function construct which I think would be strictly worse

Reply:
I don't really want two entry points to `Connector.Connect`. Both of the current options are used fairly deep in the connection logic, so I don't think we want two paths to them. 

We could, however, add a sibling to `daemon.GetClient` and see if the arguments can be split up. It's somewhat of a judgement call right now, since we haven't been living with this code yet. I think I would wait to see if or how the set of options and use cases evolve.